lib/lintPostcssResult.js

```diff
@@ -107,16 +107,27 @@ function lintPostcssResult(stylelintOptions, postcssResult, config) {
 
 		performRules.push(
 			Promise.all(
-				postcssRoots.map((postcssRoot) =>
-					ruleFunction(primaryOption, secondaryOptions, {
-						fix:
-							stylelintOptions.fix &&
-							// Next two conditionals are temporary measures until #2643 is resolved
-							isFileFixCompatible &&
-							!postcssResult.stylelint.disabledRanges[ruleName],
-						newline,
-					})(postcssRoot, postcssResult),
-				),
+				postcssRoots.map(async (postcssRoot) => {
+					try {
+						return await ruleFunction(primaryOption, secondaryOptions, {
+							fix:
+								stylelintOptions.fix &&
+								// Next two conditionals are temporary measures until #2643 is resolved
+								isFileFixCompatible &&
+								!postcssResult.stylelint.disabledRanges[ruleName],
+							newline,
+						})(postcssRoot, postcssResult);
+					} catch (ex) {
+						const fileName =
+							postcssRoot.source && postcssRoot.source.input
+								? postcssRoot.source.input.file
+								: 'unknown file';
+
+						// eslint-disable-next-line no-console
+						console.error(`Rule ${ruleName} threw an unknown exception from ${fileName}`);
```

Comment:
[IMO] I have some suggestions about this message:

- How about quoting a rule name?
- How about quoting a file name?
- How about using `<input css>` if a file name is not present?

For example:

```
Rule "indentation" threw an unknown exception from "/home/foo/button.sass"

Rule "indentation" threw an unknown exception from <input css>
```

See also:
https://github.com/stylelint/stylelint/blob/f0b5aa864e9e1563aef6a697e2dd26d3100eb946/lib/createPartialStylelintResult.js#L81